src/librustc/mir/visit.rs

```diff
@@ -248,6 +248,7 @@ macro_rules! make_mir_visitor {
                     ref $($mutability)* var_decls,
                     ref $($mutability)* arg_decls,
                     ref $($mutability)* temp_decls,
+                    upvar_decls: _,
```

Comment:
I didn't want it to be part of MIR itself, really - initially I had `upvar_names: Vec<Name>` but realized that I also needed to know if they're by-ref, that's the only reason it's a `struct`.